torch/csrc/Stream.h

```diff
@@ -10,6 +10,8 @@ struct THPStream {
   int64_t stream_id;
   int64_t device_type;
   int64_t device_index;
+  // Used to switch stream context management, initialized lazily.
+  PyObject* context;
```

Comment:
Ho this is a bit different from what I suggested haha. I think is one is ok even though it is a bit weird. The two small improvements I could see are:
1) use the functions I shared with you offline, these would used the dict objects natively managed by CPython that all PyObject already have! It matches the `__dict__` attribute you can look at from python. This will simplify your code as you don't need the new entry here in Stream.h and you won't need to manage the dictionary yourself.
2) don't define this generic dict and just add 2 entries here for device_idx and stream as two PyObject* and just manage these without putting them in a dic.

Reply:
Hi @albanD 
>>>use the functions I shared with you offline, these would used the dict objects natively managed by CPython that all PyObject already have! It matches the __dict__ attribute you can look at from python. 

`THPStream` is an immutable class. It seems that we can't access its `__dict__` by adding new items.

>>> don't define this generic dict and just add 2 entries here for device_idx and stream as two PyObject* 

`THPStream` is a generic stream class that should contain `stream_id`, `device_type`, and `device_index`. If it has two `device_index`, one is `int64_t`, another one is `PyObject*`. And even it contains a `PyObject* stream`. Do you think it is a bit weird?

Reply:
You can use https://docs.python.org/3/c-api/typeobj.html#c.Py_TPFLAGS_MANAGED_DICT in the newest versions of CPython to get this dict field easily.
But we might have to do some stuff to support older CPython.

Let's keep it as is for now and we can migrate it away in the future.